### ISSUE TITLE
tsc no longer reports a non-zero error code when reporting errors

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#4095]
Try installing the nightly with `npm install -g typescript@next` and compile the following file:

```TypeScript
var asdf = 123
asdf = '123'
```

If you try to get the error code from 20150730 (or really any nightly that's available right now), it'll be `0`. If you switch back to 1.5 with `npm install -g typescript@latest`, you'll get an error code of `2`.

**Comment by RyanCavanaugh** [comment]
Presumably this is why the DefinitelyTyped test runner didn't report errors?

**Comment by DanielRosenwasser** [comment]
Precisely - courtesy of @basarat [on Gitter](https://gitter.im/borisyankov/DefinitelyTyped?at=55baf844dcea60b379febd7b).

**Comment by mhegazy** [comment]
@DanielRosenwasser can you take a look at the issue. we should resolve this quickly to avoid more breaks in tools down stream.

**Comment by mhegazy** [comment]
thanks!
